src/domain/safe/entities/transfer.entity.ts

```diff
@@ -25,8 +41,8 @@ export interface ERC721Transfer extends Transfer {
 export function isERC721Transfer(
   transfer: Transfer,
 ): transfer is ERC721Transfer {
-  const erc721Transfer = transfer as ERC721Transfer;
-  return erc721Transfer.tokenAddress != null && erc721Transfer.tokenId != null;
+  const hasTokenId = 'tokenId' in transfer && transfer.tokenId != null;
```

Comment:
Following the logic on `isERC20Transfer` we could also extract `hasTokenId` to a function to have a clearer implementation, wdyt?

Reply:
Good idea. I've extracted the logic accordingly in 7cc62fa26e597a277f5adc5a0bda4ed3cba0e0fc.